File: code_snippets/java/src/main/java/develop/workflows/UserManagementService.java
package develop.workflows;

import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.VirtualObject;

@VirtualObject
public class UserManagementService {
  // <start_here>
  @Handler
  public void setup(ObjectContext ctx, Email email) {
    // !focus
    boolean result = SignupWorkflowClient.fromContext(ctx, "someone").run(email).await();
  }

  @Handler
  public void queryStatus(ObjectContext ctx) {
    // !focus
    String status = SignupWorkflowClient.fromContext(ctx, "someone").getStatus().await();
  }
  // <end_here>
}

File: code_snippets/kotlin/src/main/kotlin/develop/workflows/UserManagementService.kt
package develop.workflows

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.VirtualObject
import dev.restate.sdk.kotlin.ObjectContext

@VirtualObject
class UserManagementService {
  // <start_here>
  @Handler
  suspend fun setup(ctx: ObjectContext, email: Email) {
    // !focus
    val result = SignupWorkflowClient.fromContext(ctx, "someone").run(email).await()
  }

  @Handler
  suspend fun queryStatus(ctx: ObjectContext) {
    // !focus
    val status = SignupWorkflowClient.fromContext(ctx, "someone").getStatus().await()
  }
  // <end_here>
}
